aspnetcore/fundamentals/routing/samples/8.x/OptionalParams/Controllers/MyController.cs

```diff
@@ -0,0 +1,15 @@
+﻿using Microsoft.AspNetCore.Mvc;
+
+namespace WebApplication1.Controllers;
+
+public class MyController : ControllerBase
+{
+    // GET /api/my/red/2/joe
+    // GET /api/my/red/2
+    // GET /api/my
+    [Route("api/[controller]/{color}/{id:int?}/{name?}")]
```

Comment:
Should this be `HttpGet` based on the comments, so it would only match the get requests?
```suggestion
    [HttpGet("api/[controller]/{color}/{id:int?}/{name?}")]
```

Reply:
`Route` by itself will only match `GET` requests. You could argue for style points or to make it more obvious. 

`[Route("api/[controller]/{color}/{id:int?}/{name?}")]` is the same as `[HttpGet("api/[controller]/{color}/{id:int?}/{name?}")]`
So I'm inclined to leave it as is. 
It  might have slightly more SEO power with Route, then again, it might have less for some folks searching on `HttpGet optional param`.

Reply:
@Kahbazi  check [this commit](https://github.com/dotnet/AspNetCore.Docs/pull/29322/commits/84d3dcaaa8a38f7610a5ac7e5c84b33c682f55c4). I like your idea